leader_test.go

```diff
@@ -417,3 +420,107 @@ func TestLeader_nodeLists(t *testing.T) {
 			len(health), len(ping))
 	}
 }
+
+const namespacesJSON = `[
+  { "Name": "default", "Description": "Builtin Default Namespace" },
+  { "Name": "foo", "Description": "foo" }
+]`
+
+const healthserviceJSON = `[
+  { "Service": {"ID": "one", "Namespace": "foo" } },
+  { "Service": {"ID": "two", "Namespace": "default" } }
+]`
+
+func Test_namespacesList(t *testing.T) {
+	testcase := ""
+	ts := httptest.NewServer(http.HandlerFunc(
```

Comment:
This is a really cool pattern for handling the mock responses! I didn't know you could do a switch on a local variable like this in an anonymous function.

Reply:
Love that lexical scoping. :)